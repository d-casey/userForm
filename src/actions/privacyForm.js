import {
  SUBMIT_PRIVACY_FORM,
  RESET_PRIVACY_FORM
} from '../constants/actions'

export const submitPrivacyForm = formFields => ({
  type: SUBMIT_PRIVACY_FORM,
  payload: formFields
})

export const resetPrivacyForm = () => ({
  type: RESET_PRIVACY_FORM
})
