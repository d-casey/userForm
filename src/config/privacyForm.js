const privacyForm = {
  checkboxData: [
    {
      name: 'receive_updates',
      defaultChecked: true,
      message: 'Receive Updates about Tray.io products by email',
      testId: 'receive_updates'
    },
    {
      name: 'receive_communication',
      message: 'Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team',
      testId: 'receive_communication'
    }
  ]
}

export default privacyForm
