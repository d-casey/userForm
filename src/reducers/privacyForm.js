import {
  SUBMIT_PRIVACY_FORM
} from '../constants/actions'

const initialState = {
  receive_updates: true,
  receive_communication: false,
  isSubmitted: false
}

export default function newUserForm(state = initialState, action) {
  switch (action.type) {
    case SUBMIT_PRIVACY_FORM:
    console.log('payload: ', action.payload)
      return {
        ...state,
        receive_updates: action.payload.receive_updates,
        receive_communication: action.payload.receive_communication,
        isSubmitted: true
      }

    default:
      return state
  }
}
