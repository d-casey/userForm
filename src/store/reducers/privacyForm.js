import {
  RESET_PRIVACY_FORM,
  SUBMIT_PRIVACY_FORM
} from '../constants/actions'

const initialState = {
  receive_updates: true,
  receive_communication: false,
  isSubmitted: false
}

export default (state = initialState, action) => {
  switch (action.type) {
    case SUBMIT_PRIVACY_FORM:
      return {
        ...state,
        receive_updates: action.payload.receive_updates,
        receive_communication: action.payload.receive_communication,
        isSubmitted: true
      }

      case RESET_PRIVACY_FORM:
        return initialState

    default:
      return state
  }
}
