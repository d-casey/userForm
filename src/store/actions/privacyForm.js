import {
  SUBMIT_PRIVACY_FORM,
  RESET_PRIVACY_FORM
} from '../constants/actions'

import { updateStage } from '../actions'

export const updatePrivacyFields = formFields => ({
  type: SUBMIT_PRIVACY_FORM,
  payload: formFields
})

export const submitPrivacyForm = formFields => dispatch => {
  dispatch(updatePrivacyFields(formFields))
  dispatch(updateStage(2))
}

export const resetPrivacyForm = () => ({
  type: RESET_PRIVACY_FORM
})
