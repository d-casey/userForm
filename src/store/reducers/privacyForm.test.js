import {
  RESET_PRIVACY_FORM,
  SUBMIT_PRIVACY_FORM
} from '../constants/actions'

import privacyForm from './privacyForm'

test('it updates the privacy form fields correctly', () => {
  expect(privacyForm({},
      {
        type: SUBMIT_PRIVACY_FORM,
        payload:
        [
          {
            name: 'receive_updates',
            defaultChecked: true,
            message: 'Receive Updates about Tray.io products by email'
          },
          {
            name: 'receive_communication',
            message: 'Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team'
          }
        ]
      }
    )
  )
  .toMatchSnapshot()
})

test('it resets the stage correctly', () => {
  expect(privacyForm({}, { type: RESET_PRIVACY_FORM })).toMatchSnapshot()
})

test('it returns the default state when nothing provided', () => {
  expect(privacyForm({}, { type: 'UNKNOWN' })).toMatchSnapshot()
})
