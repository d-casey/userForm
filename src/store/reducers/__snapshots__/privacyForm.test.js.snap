// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it resets the stage correctly 1`] = `
Object {
  "isSubmitted": false,
  "receive_communication": false,
  "receive_updates": true,
}
`;

exports[`it returns the default state when nothing provided 1`] = `Object {}`;

exports[`it updates the privacy form fields correctly 1`] = `
Object {
  "0": Object {
    "defaultChecked": true,
    "message": "Receive Updates about Tray.io products by email",
    "name": "receive_updates",
  },
  "1": Object {
    "message": "Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team",
    "name": "receive_communication",
  },
  "isSubmitted": true,
}
`;
